llvm_mode/afl-llvm-pass.so.cc

```diff
@@ -104,36 +118,38 @@ bool AFLCoverage::runOnModule(Module &M) {
      __afl_prev_loc is thread-local. */
 
   GlobalVariable *AFLMapPtr =
-      new GlobalVariable(M, PointerType::get(Int8Ty, 0), false,
+      new GlobalVariable(M, PointerType::get(Int64Ty, 0), false,
                          GlobalValue::ExternalLinkage, 0, "__afl_area_ptr");
 
   GlobalVariable *AFLPrevLoc = new GlobalVariable(
-      M, Int32Ty, false, GlobalValue::ExternalLinkage, 0, "__afl_prev_loc",
+      M, Int64Ty, false, GlobalValue::ExternalLinkage, 0, "__afl_prev_loc",
       0, GlobalVariable::GeneralDynamicTLSModel, 0, false);
 
   /* Instrument all the things! */
 
   int inst_blocks = 0;
-
+  u64 block_counter = 0;
+  std::ofstream location_file;
+  std::error_code llvm_of_error; 
+  
   for (auto &F : M)
     for (auto &BB : F) {
 
+      assert(block_counter < MAP_SIZE && "counter is too large");
       BasicBlock::iterator IP = BB.getFirstInsertionPt();
       IRBuilder<> IRB(&(*IP));
 
       if (AFL_R(100) >= inst_ratio) continue;
```

Comment:
Let's also remove this, we don't need it. It's an AFL thing only.

Reply:
Looking at the code a second time, it looks like we don't need many of the loads as well, since the counter is constant.